contracts/MechMarketplace.sol

```diff
@@ -321,16 +358,29 @@ contract MechMarketplace is IErrorsMarketplace {
         // Increase the amount of requester delivered requests
         mapDeliveryCounts[requester]++;
         // Increase the amount of mech service multisig delivered requests
-        mapMechServiceDeliveryCounts[mechService]++;
+        mapMechServiceDeliveryCounts[mechServiceMultisig]++;
+
+        // Record request counts in global sets
+        IMechManager(mechManager).increaseDeliveryCounts(requester, msg.sender, mechServiceMultisig);
 
         // Increase mech karma that delivers the request
-        IKarma(karmaProxy).changeMechKarma(msg.sender, 1);
+        IKarma(karma).changeMechKarma(msg.sender, 1);
+
+        // Process payment
+        _processPayment(msg.sender, requestId);
 
         emit MarketplaceDeliver(priorityMech, msg.sender, requester, requestId, requestData);
 
         _locked = 1;
     }
 
+    // TODO: send to BBB, wrap first or manage on BBB side?
+    function drain() external {
+        if (collectedFees == 0) {
+            revert ZeroValue();
+        }
+    }
```

Comment:
BBB knows nothing about Marketplace and only accepts OLAS or native (wrapped). So we should send the wrapped version.